### ISSUE TITLE
torch.compile CPU backend is slower than eager for several transcendental functions

### ISSUE THREAD

**Issue by bertmaher** [pytorch/pytorch#110611]
I tried a handful of somewhat important transcendentals (sigmoid, tanh, log, logit), and they all appear to be slower than eager.  On my 96c/192t Intel(R) Xeon(R) Platinum 8339HC CPU:

```
$ OMP_NUM_THREADS=1 numactl -C 94 python bench.py
log
eager:    363.3282147347927
compiled: 1091.864187270403
sigmoid
eager:    756.7794565111399
compiled: 872.3638635128736
logit
eager:    731.156948953867
compiled: 1594.5508629083633
tanh
eager:    475.9031366556883
compiled: 867.8419776260853

$ python bench.py
log
eager:    28.686294332146645
compiled: 106.62426799535751
sigmoid
eager:    91.70412458479404
compiled: 108.63539576530457
logit
eager:    25.92829428613186
compiled: 88.3821751922369
tanh
eager:    28.41925248503685
compiled: 113.94776217639446
```

```
import time

import torch

def do_bench(fn, iters=1000):
    fn()  # warmup
    s = time.perf_counter()
    for _ in range(iters):
        fn()
    e = time.perf_counter()

    return (e - s) /  iters * 1e6

x = torch.clamp(torch.rand(1 << 20), 1e-6, 1 - 1e-6)  # good range for log, logit

for fn in [torch.log,
           torch.sigmoid,
           torch.logit,
           torch.tanh,
           ]:
    cfn = torch.compile(fn)
    print(fn.__name__)

    eager = do_bench(lambda: fn(x))
    compiled = do_bench(lambda: cfn(x))

    print(f"eager:    {eager}")
    print(f"compiled: {compiled}")
```

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @aakhundov @ColinPeppler

**Comment by eellison** [comment]
cc @jgong5

**Comment by jgong5** [comment]
There are two issues:
1. Inductor vectorizes `log` with `Sleef_logf16_u10` which is slower than MKL VML function used by the eager implementation. This impacts the perf of `torch.log` and `torch.logit`. It is similar to the perf issue for erf, sin and cos that @Valentine233 fixed in https://github.com/pytorch/pytorch/pull/91613 and https://github.com/pytorch/pytorch/pull/94577. Using `Sleef_logf16_u35` brings the performance back but needs to check the accuracy impact. Assigned to @Valentine233 to optimize log too.

The original perf number:
```bash
(pytorch) jgong5@mlp-icx-06:~/pytorch$ numactl -C0 python ../tmp/test.py
log
eager:    368.8463559374213
compiled: 616.8672097846866
sigmoid
eager:    423.1066880747676
compiled: 461.9728191755712
logit
eager:    565.499295014888
compiled: 1010.4096410796046
tanh
eager:    352.703922893852
compiled: 489.6416366100311
```

Perf with `Sleef_logf16_u35`
```bash
(pytorch) jgong5@mlp-icx-06:~/tmp$ numactl -C0 python test.py
log
eager:    364.8629770614207
compiled: 360.2141812443733
sigmoid
eager:    433.7635678239167
compiled: 461.424479726702
logit
eager:    562.3160391114652
compiled: 545.2622110024095
tanh
eager:    351.78632382303476
compiled: 490.19992584362626
```

2. torch.compile overhead
With 1 socket (32 cores) and the optimization in 1, I got:
```bash
(pytorch) jgong5@mlp-icx-06:~/tmp$ numactl -C0-31 python test.py
log
eager:    34.285707864910364
compiled: 67.15910695493221
sigmoid
eager:    29.539030976593494
compiled: 70.79279189929366
logit
eager:    31.614462845027447
compiled: 74.05219273641706
tanh
eager:    26.97620214894414
compiled: 73.41079507023096
```
If we look into sigmoid profile, we get:
```bash
(pytorch) jgong5@mlp-icx-06:~/tmp$ numactl -C0-31 python test.py
sigmoid
-----------------  ------------  ------------  ------------  ------------  ------------  ------------  
             Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg    # of Calls  
-----------------  ------------  ------------  ------------  ------------  ------------  ------------  
    ProfilerStep*         6.07%       2.932ms       100.00%      48.312ms      48.312ms             1  
    aten::sigmoid        93.93%      45.380ms        93.93%      45.380ms      45.380us          1000  
-----------------  ------------  ------------  ------------  ------------  ------------  ------------  
Self CPU time total: 48.312ms

-------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  
                           Name    Self CPU %      Self CPU   CPU total %     CPU total  CPU time avg    # of Calls  
-------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  
                  ProfilerStep*        16.07%      11.857ms       100.00%      73.786ms      73.786ms             1  
       TorchDynamo Cache Lookup         4.34%       3.199ms         4.34%       3.199ms       3.199us          1000  
          Torch-Compiled Region        23.23%      17.141ms        79.60%      58.730ms      58.730us          1000  
            aten::empty_strided         0.07%      52.000us         0.07%      52.000us       0.052us          1000  
    graph_0_cpp_fused_sigmoid_0        56.29%      41.537ms        56.29%      41.537ms      41.537us          1000  
-------------------------------  ------------  ------------  ------------  ------------  ------------  ------------  
Self CPU time total: 73.786ms

eager:    48.26375097036362
compiled: 73.73075187206268
```
The kernel time is on par with `aten::sigmoid`, which is expected since the both implementations use aten vec and are similar. But there is also overhead with torch.compile in the python wrapper and the `Torch-Compiled Region`. Below is the chrome-trace with cpp wrapper which eliminates the overhead in python wrapper but still a bit overhead in `Torch-Compiled Region` due to dynamo. This is probably benign for large graph with many ops though.
![image](https://github.com/pytorch/pytorch/assets/8359223/5186fa79-a5cd-414a-8cf9-a669b9d8dd06)

**Comment by Valentine233** [comment]
The PR to replace `Sleef_logf16_u10` with `Sleef_logf16_u35` was reverted for accuracy issues.

We decide to re-implement the kernel later.

**Comment by Valentine233** [comment]
Need to re-check the current status, as the inductor compiling overhead was optimized.
